hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -262,16 +274,28 @@ object DefaultSource {
           new IncrementalRelation(sqlContext, parameters, userSchema, metaClient)
 
         case (MERGE_ON_READ, QUERY_TYPE_SNAPSHOT_OPT_VAL, false) =>
-          new MergeOnReadSnapshotRelation(sqlContext, parameters, metaClient, globPaths, userSchema)
+          if (newHudiFileFormatUtils.isEmpty || newHudiFileFormatUtils.get.hasSchemaOnRead) {
```

Comment:
Instead of checking `newHudiFileFormatUtils.get.hasSchemaOnRead`, should `hoodie.schema.on.read.enable` config be checked when lazy creating `newHudiFileFormatUtils` and only check `newHudiFileFormatUtils.isEmpty` here for simplicity?